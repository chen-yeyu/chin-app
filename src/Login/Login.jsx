import React, { Component, PropTypes } from 'react';
import {Provider, connect} from 'react-redux';
import {createStore, combineReducers} from 'redux';

const FETCH_ACTION = "fetch send";
const FETCH_FAIL = "fetch fail";
const CLEAN_SCREEN = "clean screen";

const FetchAction = function(value){
  return {
    type: FETCH_ACTION,
    value
  }
}

const FetchFail = function(value){
  return {
    type: FETCH_FAIL,
    value
  }
}

const CleanScreenAction = function(value){
  return {
    type: CLEAN_SCREEN,
    value
  }
}

function CleanScreen () {
  var data = {id:"", username:"", password:"", last_name:"", first_name:"", message:""};
  LoginStore.dispatch(CleanScreenAction(data));
}

//localhost:3030/test2
function FetchPosts (value) {
  var mydata = value;
  fetch('http://localhost:5001/mypage/react', {
    method : 'POST',
    mode : 'cors',
    body: mydata
  })
  .then(function(response){
    return response.json();
  })
  .then(function(data){
    LoginStore.dispatch(FetchAction(data));
  })
  .catch(function(e){
    LoginStore.dispatch(FetchFail({message: e.message}));
  })
}

const LoginReducer = function(state={}, action){
  switch (action.type) {
    case FETCH_ACTION:
      return Object.assign({}, state, action.value);
    case FETCH_FAIL:
      return Object.assign({}, state, action.value);
    case CLEAN_SCREEN:
      return Object.assign({}, state, action.value);
    default:
      return state;

  }
}

// const LoginFailReducer = function(state={}, action){
//   switch (action.type) {
//     case FETCH_FAIL:
//       return Object.assign({}, state, action.value);
//     default:
//       return state;
//
//   }
// }

class App extends Component {

  constructor(props,context){
    super(props, context);
    this.clickLogin = this.clickLogin.bind(this);
  }

  clickLogin(){
    CleanScreen();
    var formData = new FormData();
    formData.append('username', this.refs.username.value);
    formData.append('password', this.refs.password.value);
    FetchPosts(formData);
  }

  render() {
    return (
      <div className="App">
        <h1>username: </h1><input type="text" ref="username"/><br/>
        <h1>password: </h1><input type="password" ref="password"/><br/>
        <br/>
        <button onClick={this.clickLogin} style={{width: 300, height:100}}>Login</button>
        <br/>
        <h1>user id:{this.props.id == undefined || "" ? "  no user" : this.props.id}</h1>
        <h1>user password:{this.props.password}</h1>
        <h1>user username:{this.props.username}</h1>
        <h1>user first_name:{this.props.first_name}</h1>
        <h1>user last_name:{this.props.last_name}</h1>
        <h1>post error:{this.props.message}</h1>
      </div>
    );
  }
}

const mapStateToProps = function(state){
  return state;
};

//const rootReducer = combineReducers({LoginReducer, LoginFailReducer});

const LoginStore = createStore(LoginReducer);

const ConnectedApp = connect(mapStateToProps)(App);

export default class Login extends Component {
  render(){
    return(<Provider store={LoginStore}><ConnectedApp /></Provider>);
  }

}
