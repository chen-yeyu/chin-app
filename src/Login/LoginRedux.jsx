import React, { Component } from 'react';
import { Provider , connect } from 'react-redux'
import { createStore, applyMiddleware, combineReducers } from 'redux'
import { Field, reduxForm, reducer as formReducer, SubmissionError} from 'redux-form'

const FETCH_ACTION = "fetch send";
const FETCH_FAIL = "fetch fail";
const CLEAN_SCREEN = "clean screen";

const FetchAction = function(value){
  return {
    type: FETCH_ACTION,
    value
  }
}

const FetchFail = function(value){
  return {
    type: FETCH_FAIL,
    value
  }
}

const CleanScreenAction = function(value){
  return {
    type: CLEAN_SCREEN,
    value
  }
}

function CleanScreen () {
  var data = {id:"", username:"", password:"", last_name:"", first_name:"", message:""};
  LoginStore.dispatch(CleanScreenAction(data));
}

function FetchPosts (value) {
  var mydata = value;
  fetch('http://localhost:5001/mypage/', {
    method : 'POST',
    mode : 'cors',
    body: mydata
  })
  .then(function(response){
    return response.json();
  })
  .then(function(data){
    LoginStore.dispatch(FetchAction(data));
  })
  .catch(function(e){
    LoginStore.dispatch(FetchFail({message: e.message}));
  })
}

const LoginReducer = function(state={}, action){
  switch (action.type) {
    case FETCH_ACTION:
      return Object.assign({}, state, action.value);
    case FETCH_FAIL:
      return Object.assign({}, state, action.value);
    case CLEAN_SCREEN:
      return Object.assign({}, state, action.value);
    default:
      return state;

  }
}

class validateForm extends Component {
  render(){
    const {input, label, type, inline,title, meta: { asyncValidating, touched, error }} = this.props;
    return (
      <span>
        {touched && error && <span style={{"color":"green"}}> {title} {error} </span>}<br/>
        <label>{label}</label> <input {...input} className="corp-ReduxThunk-form-field" type={type} placeholder={label} />
     </span>
    );

  }

}

class loginForm extends Component {

  initialValues:{
  username: "",
  password: ""
  }

  render(){
    let this_form = this.props;
    return (<form>
            <div>
                <Field name="username" type="text" component={validateForm} label="username: " title = "ユーザ名:"/><br/>
                <Field name="password" type="password" component={validateForm} label="password: " title = "パスワード:"/>
           </div>
         </form>);
  }
}

class App extends Component {
  render(){
    return (
      <form>
        <Field name="login" component={loginForm} />
        <button type="submit" onClick={this.props.handleSubmit(clickLogin(this.props))}> Login </button><br/>
        <h1>user id:{this.props.LoginReducer.id == undefined || this.props.LoginReducer.id == "" ? "  no user" : this.props.LoginReducer.id}</h1>
        <h1>user password:{this.props.LoginReducer.password}</h1>
        <h1>user username:{this.props.LoginReducer.username}</h1>
        <h1>user first_name:{this.props.LoginReducer.first_name}</h1>
        <h1>user last_name:{this.props.LoginReducer.last_name}</h1>
        <h1>post error:{this.props.LoginReducer.message}</h1>
      </form>
    );
  }
}

const validate = values => {
  const errors = {}
  if (!values.username) {
    errors.username = 'Required';
  } else if (values.username.length > 3) {
    errors.username = 'Must be 3 characters or less';
  }

  if(!values.password) {
    errors.password = "Required";
  }else if (values.password.length > 3) {
    errors.password = 'Must be 3 characters or less';
  }
  return errors
}

const LoginState = function(state){
  return state;
}

const reduxLoginForm = reduxForm({
  form: "sample",
  validate
})(App)

const LoginForm = connect(LoginState)(reduxLoginForm)

const RootReducer = combineReducers({LoginReducer, form:formReducer});

const LoginStore = createStore(RootReducer);



class LoginRedux extends React.Component{
  render(){
    return <Provider store={LoginStore}><LoginForm /></Provider>
  }
}

export const clickLogin = (props) => (values) => {
  CleanScreen();
  var formData = new FormData();
  formData.append('username', values.username);
  formData.append('password', values.password);
  FetchPosts(formData);
}

export default LoginRedux;
