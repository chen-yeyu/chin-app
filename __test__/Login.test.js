import 'whatwg-fetch'
import React from 'react';
import Login from '../src/Login/Login';
import * as LoginAction from '../src/Login/Login';
import renderer from 'react-test-renderer';

describe('component', () =>{

  test('matches snapshot', () => {
    const tree = renderer.create(
      <Login />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

})

describe('actioin', () =>{

  it('works with resolves', () => {
    let result = LoginAction.getUserName();
    expect(result).toEqual(5)
  });
})

const TestFetch = function(url , data , callback){
  fetch(url, {
    method: 'POST',
    mode: 'cors',
    body: data
  }).then(callback)
}

describe('async action', () => {

  test('fetchPosts test', () => {
    fetch('http://localhost:5001/mypage/react',{
      method : 'POST',
      mode: 'cors',
      body: {username:"cyy", password:123}
    }).then(function(res){
      expect(res).toEqual("object");
    })
  })
})


describe('TestFetch action', () => {

  test('TestFetch test', () => {
    var url = "http://localhost:5001/mypage/react";
    var data = {username:"cyy",password:123};
    TestFetch(url , data , )
  })
})
