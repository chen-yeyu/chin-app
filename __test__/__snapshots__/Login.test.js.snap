// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component matches snapshot 1`] = `
<div
  className="App"
>
  <h1>
    username: 
  </h1>
  <input
    type="text"
  />
  <br />
  <h1>
    password: 
  </h1>
  <input
    type="password"
  />
  <br />
  <br />
  <button
    onClick={[Function]}
    style={
      Object {
        "height": 100,
        "width": 300,
      }
    }
  >
    Login
  </button>
  <br />
  <h1>
    user id:
      no user
  </h1>
  <h1>
    user password:
  </h1>
  <h1>
    user username:
  </h1>
  <h1>
    user first_name:
  </h1>
  <h1>
    user last_name:
  </h1>
  <h1>
    post error:
  </h1>
</div>
`;

exports[`matches snapshot 1`] = `
<div
  className="App"
>
  <h1>
    username: 
  </h1>
  <input
    type="text"
  />
  <br />
  <h1>
    password: 
  </h1>
  <input
    type="password"
  />
  <br />
  <br />
  <button
    onClick={[Function]}
    style={
      Object {
        "height": 100,
        "width": 300,
      }
    }
  >
    Login
  </button>
  <br />
  <h1>
    user id:
      no user
  </h1>
  <h1>
    user password:
  </h1>
  <h1>
    user username:
  </h1>
  <h1>
    user first_name:
  </h1>
  <h1>
    user last_name:
  </h1>
  <h1>
    post error:
  </h1>
</div>
`;
